test/Dialect/HW/errors.mlir

```diff
@@ -26,7 +26,7 @@ func private @notModule () {
 }
 
 hw.module @A(%arg0: i1) {
-  // expected-error @+1 {{symbol reference 'notModule' isn't a module}}
+  // expected-error @+1 {{'hw.instance' op attribute 'moduleName' failed to satisfy constraint: flat symbol reference attribute is module like}}
```

Comment:
Curiously, is there any kind of Filecheck support to make this less brittle by looking at a substring of the message

Reply:
Actually, this isn't filecheck -- it's built into the llvm executable and run with -verify-diagnostics I think. Also, I don't see a sufficiently descriptive common substring, so we'd probably need regex support -- which filecheck has.

Reply:
The error message should be fairly exact.